toqito/matrices/operator_sinkhorn.py

```diff
@@ -0,0 +1,209 @@
+"""Sinkhorn operator."""
+
+import warnings
+
+import numpy as np
+import scipy as sp
+
+from toqito.channels.partial_trace import partial_trace
+from toqito.matrix_props.is_square import is_square
+
+
+def operator_sinkhorn(
+    rho: np.ndarray, dim: list[int] = None, tol: float = np.sqrt(np.finfo(float).eps), max_iterations: int = 10000
+) -> tuple[np.ndarray, list[np.ndarray]]:
+    r"""Perform operator Sinkhorn iteration (iterative sinkhorn algorithm) on a quantum system's density matrix.
+
+    This function relies on Sinkhorn's theorem :cite:`Sinkhorn_1964_ARelationship` which states "for any
```

Comment:
Somewhere in this we should be citing QETLAB to indicate that the reference implementation upon which we are basing it is coming from that implementation.